cfme/utils/log_validator.py

```diff
@@ -46,16 +46,23 @@ def __init__(self, remote_filename, **kwargs):
         self._remote_file_tail = SSHTail(remote_filename, **kwargs)
         self.matches = {}
 
+    def update_matched_patterns(self, new_patterns):
+        self.matched_patterns = new_patterns
+        self.matches = {}
+
     def fix_before_start(self):
         self._remote_file_tail.set_initial_file_end()
 
-    def validate_logs(self):
+    def validate_logs(self, _auto_fail=True):
         for line in self._remote_file_tail:
             if self._check_skip_logs(line):
                 continue
             self._check_fail_logs(line)
             self._check_match_logs(line)
-        self._verify_match_logs()
+        if _auto_fail:
+            self._verify_match_logs()
+        else:
+            return list(self.matches.keys())
```

Comment:
We can always return, no need to put it in else. Also, can go with `return list(self.matches)` here as well.